spark/src/test/scala/org/apache/spark/sql/delta/CheckpointsSuite.scala

```diff
@@ -37,9 +37,43 @@ import org.apache.spark.sql.{QueryTest, Row}
 import org.apache.spark.sql.test.SharedSparkSession
 import org.apache.spark.sql.types.StructType
 
+trait CheckpointsSuiteBase extends QueryTest with SharedSparkSession {
+  def testDifferentV2Checkpoints(testName: String)(f: => Unit): Unit = {
+    for (checkpointFormat <- Seq(V2Checkpoint.Format.JSON.name, V2Checkpoint.Format.PARQUET.name)) {
+      test(s"$testName [v2CheckpointFormat: $checkpointFormat]") {
+        withSQLConf(
+          DeltaConfigs.CHECKPOINT_POLICY.defaultTablePropertyKey -> CheckpointPolicy.V2.name,
+          DeltaSQLConf.CHECKPOINT_V2_TOP_LEVEL_FILE_FORMAT.key -> checkpointFormat
+        ) {
+          f
+        }
+      }
+    }
+  }
+
+  /** Get V2 [[CheckpointProvider]] from the underlying deltalog snapshot */
+  def getV2CheckpointProvider(
+      deltaLog: DeltaLog,
+      update: Boolean = true): V2CheckpointProvider = {
+    val snapshot = if (update) deltaLog.update() else deltaLog.unsafeVolatileSnapshot
+    snapshot.checkpointProvider match {
+      case v2CheckpointProvider: V2CheckpointProvider =>
+        v2CheckpointProvider
+      case lzProvider : LazyCompleteCheckpointProvider
+          if lzProvider.underlyingCheckpointProvider.isInstanceOf[V2CheckpointProvider] =>
+        lzProvider.underlyingCheckpointProvider.asInstanceOf[V2CheckpointProvider]
```

Comment:
cryptic abbreviations don't really help IMO...
```suggestion
      case provider : LazyCompleteCheckpointProvider
          if provider.underlyingCheckpointProvider.isInstanceOf[V2CheckpointProvider] =>
        provider.underlyingCheckpointProvider.asInstanceOf[V2CheckpointProvider]
```
(meanwhile, the cast is annoying, but I'm not sure it's annoying enough to be worth making an extractor for)